generator/enum_atom.py

```diff
@@ -1,45 +1,39 @@
-'''
-Generate enum.h file.
-Run this file (with the spec path as a command line argument) to write just
-ids.h or main.py to write all files.
-'''
 import sys
 sys.path.append("ParseCAN")
 import ParseCAN
-from common import enum_atom_path, templ, coord, ifndef, endif
+from common import enum_atom_path, templ, coord, ifndef, endif, frame_handler
 
 
-def write(can, output_path=enum_atom_path):
-    '''
-    Generate enum_atom.h file, which includes enums for atom of type enum.
+def handle_frame(frame, name_prepends, fw):
+    for atom in frame.atom:
+        tot_name = coord(name_prepends, frame.name, atom.name)
+        if atom.type.isenum():
+            # Only C++11 feature
+            # fw('typedef enum ' + (atom.type.type + ' ' if atom.type.type else '') + '{\n')
+
+            fw('typedef enum {\n')
+
+            for enum in atom.type.enum:
+                fw(templ['enum'].format(
+                    tot_name + '_' + enum.name,
+                    enum.value
+                ))
+            
+            fw('\t' + tot_name + '_' + 'NUM_FIELDS\n')
```

Comment:
If this is not used elsewhere can we get rid of it or add a check for an enumerator with that name?

Reply:
I just want to have this available as part of the enum for vcu, and I don't think it's that uncommon. But yeah I'll add a check for that.